connectors/jdbc/translator-jdbc/src/main/java/org/teiid/translator/jdbc/exasol/ExasolExecutionFactory.java

```diff
@@ -0,0 +1,405 @@
+        registerFunctionModifier(SourceSystemFunctions.LCASE, new AliasModifier(LOWER));
+
+        //Date
+        addPushDownFunction(EXASOL, ADD_DAYS, DATE, DATE, INTEGER);
+        addPushDownFunction(EXASOL, ADD_DAYS, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, ADD_HOURS, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, ADD_MINUTES, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, ADD_MONTHS, DATE, DATE, INTEGER);
+        addPushDownFunction(EXASOL, ADD_MONTHS, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, ADD_SECONDS, TIMESTAMP, TIMESTAMP, DOUBLE);
+        addPushDownFunction(EXASOL, ADD_WEEKS, DATE, DATE, INTEGER);
+        addPushDownFunction(EXASOL, ADD_WEEKS, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, ADD_YEARS, DATE, DATE, INTEGER);
+        addPushDownFunction(EXASOL, ADD_YEARS, TIMESTAMP, TIMESTAMP, INTEGER);
+        addPushDownFunction(EXASOL, DATE_TRUNC, DATE, STRING, DATE);
+        addPushDownFunction(EXASOL, DATE_TRUNC, TIMESTAMP, STRING, TIMESTAMP);
+        addPushDownFunction(EXASOL, DAYS_BETWEEN, INTEGER, DATE, DATE);
+        addPushDownFunction(EXASOL, DAYS_BETWEEN, INTEGER, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, HOURS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, MINUTES_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, MONTHS_BETWEEN, DOUBLE, DATE, DATE);
+        addPushDownFunction(EXASOL, MONTHS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, POSIX_TIME, DOUBLE, TIMESTAMP);
+        addPushDownFunction(EXASOL, ROUND, DATE, DATE);
+        addPushDownFunction(EXASOL, ROUND, DATE, DATE, STRING);
+        addPushDownFunction(EXASOL, ROUND, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, ROUND, TIMESTAMP, TIMESTAMP, STRING);
+        addPushDownFunction(EXASOL, SECONDS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, TRUNC, DATE, DATE);
+        addPushDownFunction(EXASOL, TRUNC, DATE, DATE, STRING);
+        addPushDownFunction(EXASOL, TRUNC, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, TRUNC, TIMESTAMP, TIMESTAMP, STRING);
+        addPushDownFunction(EXASOL, TRUNCATE, DATE, DATE);
+        addPushDownFunction(EXASOL, TRUNCATE, DATE, DATE, STRING);
+        addPushDownFunction(EXASOL, TRUNCATE, TIMESTAMP, TIMESTAMP);
+        addPushDownFunction(EXASOL, TRUNCATE, TIMESTAMP, TIMESTAMP, STRING);
+        addPushDownFunction(EXASOL, YEARS_BETWEEN, DOUBLE, DATE, DATE);
+        addPushDownFunction(EXASOL, YEARS_BETWEEN, DOUBLE, TIMESTAMP, TIMESTAMP);
+
+        registerFunctionModifier(SourceSystemFunctions.FROM_UNIXTIME, new AliasModifier(FROM_POSIX_TIME));
+        registerFunctionModifier(SourceSystemFunctions.DAYOFMONTH, new AliasModifier(DAY));
+        registerFunctionModifier(SourceSystemFunctions.CURDATE, new AliasModifier(UPPER));
```

Comment:
This function modifier is incorrect.  Generally CURDATE would be evaluated in the engine, but this should be corrected regardless.

Reply:
Right!